templates/baremetal_deployment.yaml.j2

```diff
@@ -75,6 +75,18 @@
   defaults:
     profile: baremetal{{ node_type }}
     network_config:
+{# workaround for r640's that has different interface namings #}
+{% if node_type|string() == "r640" %}
+      net_config_data_lookup:
+{% for i in range(0, ocpinventory_content.nodes | length) %}
+{% if node_type|string() in ocpinventory_content.nodes[i].pm_addr|string() %}
+        node{{ i|int + 1 }}:
+{% for j in range(0, ocpinventory_content.nodes[i].mac | length) %}
+          nic{{ j|int + 1 }}: "{{ ocpinventory_content.nodes[i].mac[j] }}"
+{% endfor %}
+{% endif %}
+{% endfor %}
+{% endif %}
       template: /home/stack/virt/network/vlans/compute_{{ node_type }}.j2
```

Comment:
How are the interface names getting updated in the template files(like compute_r640.j2)?

Reply:
[1]nic config files are updated with generic interface names like (nic1, nic2, nic3, and ni4).
[2]And in the baremetal deployment file right os-net-mappings are added so that we are matching generic interfaces names with corresponding MAC addresses.

[1]  https://github.com/redhat-performance/jetpack/pull/528/commits/e93bd60be7ec0026df7c56609889ace3646e2b5e#diff-8eaa3bbb48cc4ddaad155db5f6c17d86676ed3b8d5a8f3fa39773bbcf6adb818R78-R83
[2] https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/17.0/html/director_installation_and_usage/assembly_configuring-overcloud-networking_installing-director-on-the-undercloud#ref_configuring-custom-interfaces_additional-network-configuration